hoot-core/src/main/cpp/hoot/core/elements/Tags.cpp

```diff
@@ -235,6 +236,29 @@ QStringList Tags::getNames() const
   return result;
 }
 
+QStringList Tags::getNamesRecursive() const
+{
```

Comment:
That change was made based of your comment in the the 5722 redmine ticket about using OsmSchema::isAncestor("abstract_name") with DuplicateNameRemover instead of just alt_name as it was previously using.  Its possible, though, that I misinterpreted what you were intending.  I do see now that getNamesRecursive would be reducing the the names returned than the previous code...so maybe needs changed.